pinot-core/src/main/java/org/apache/pinot/core/operator/combine/merger/TimeSeriesAggResultsBlockMerger.java

```diff
@@ -44,6 +49,14 @@ public void mergeResultsBlocks(TimeSeriesResultsBlock mergedBlock, TimeSeriesRes
       BaseTimeSeriesBuilder newTimeSeriesToMerge = entry.getValue();
       if (currentTimeSeriesBuilder == null) {
         currentTimeSeriesBlock.getSeriesBuilderMap().put(seriesHash, newTimeSeriesToMerge);
+        final long currentUniqueSeries = currentTimeSeriesBlock.getSeriesBuilderMap().size();
+        final long numBuckets = currentTimeSeriesBlock.getTimeBuckets().getNumBuckets();
+        Preconditions.checkState(currentUniqueSeries * numBuckets <= _maxDataPointsLimit,
+            "Max data points limit reached in combine operator. Limit: %s. Current count: %s",
+            _maxDataPointsLimit, currentUniqueSeries * numBuckets);
```

Comment:
Is the behavior that when maxSeries or dataPoint limit is hit query will return failure?

Reply:
Yeah failure in both cases